ecs-init/docker/docker.go

```diff
@@ -427,6 +442,49 @@ func getDockerPluginDirBinds() []string {
 	return pluginBinds
 }
 
+// take pathPredicate as an argument for unit testing
+func getCapabilityExecBinds(pathPredicate func(string, func(os.FileInfo) bool) (bool, error)) []string {
+	hostResourcesDir := filepath.Join(hostCapabilitiesResourcesRootDir, capabilityExecName)
+	containerResourcesDir := filepath.Join(containerCapabilitiesResourcesRootDir, capabilityExecName)
+
+	var binds []string
+
+	// bind mount the entire /host/dependency/path/exec/bin folder for higher flexibility
+	// minimal change required to add other ssm binaries as dependency in the future (just need to be placed inside the bin directory)
+	hostBinDir := filepath.Join(hostResourcesDir, capabilityExecHostBinRelativePath)
```

Comment:
I think we should do individual bind mounts. If the bin directory exists but is empty, this code would say we're all good, when we're not.

Reply:
current approach (mounting `/bin`, `/certs`, etc.) is to keep all the capability detection logic on `ecs-agent` side, so it would be more flexible on the `ecs-init` side: all the actual binaries can be verified on the `ecs-agent` side (where it is also easier to share constants for binaries names, etc.). individual bind mounts, `ecs-init` should still always say we're good (launch `ecs-agent` container) since having ssm-required resources is optional (even if we don't, it shouldn't fail the ecs service)

this way, even if we update the validation logic (e.g. require more binaries), users only need to update `ecs-agent`. and `ecs-init` won't need to be added. actually similar idea for `/certs`, but I think we would always require only 1 cert file so I limited it to only 1 (the filename can also be moved to `ecs-agent` part as well)

Reply:
I see, you're right. ecs agent will have this responsibility